### PR TITLE
deps: bump to k8s 1.24

### PR DESCRIPTION
**Pull request by fromanirh**
bump to:
- k8s 1.24.7
- deployer 0.8.0
- RTE 0.8.1

Signed-off-by: Francesco Romani <fromani@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
/hold
missing bump of OCP deps (stuck around 4.10)

File: go.mod [hunk 1]
```diff
@@ -9,27 +9,27 @@ require (
...
 	github.com/onsi/ginkgo v1.16.5
 	github.com/onsi/gomega v1.17.0
 	github.com/openshift/api v0.0.0-20210924154557-a4f696157341
 	github.com/openshift/machine-config-operator v0.0.1-0.20211105081319-76d6155c1dab
 	github.com/pkg/errors v0.9.1
-	github.com/stretchr/testify v1.7.0
+	github.com/stretchr/testify v1.8.0
 	google.golang.org/grpc v1.40.0
-	k8s.io/api v0.23.3
+	k8s.io/api v0.24.7
```
Comment (swatisehgal) [inline go.mod]: What is reason behind not updating to the latest kube version i.e. 1.25.3 ?

Comment (fromanirh) [reply go.mod]: two reasons:
- minor: I want to have incremental steps and not just one big jump
- major: kubernetes 1.25+ pulls ginkgo v2, and we need to adapt our suite. Upgrading to 1.24 is cheaper and moves us in a better position for this much larger update.

Comment (swatisehgal) [reply go.mod]: Ack, makes sense. Thanks for the explanation.

**Review by swatisehgal — APPROVED** [review]
/lgtm

**Pull request merged**